tensorboard/webapp/metrics/store/metrics_store_internal_utils_test.ts

```diff
@@ -646,6 +647,173 @@ describe('metrics store utils', () => {
     });
   });
 
+  describe('generateNextCardStepIndexFromSelectedTime', () => {
+    const imageCardId = 'test imagee card id "plugin":"images"';
+    const previousCardStepIndex = {[imageCardId]: null};
+    const cardMetadataMap = {
+      [imageCardId]: {
+        runId: 'test run Id',
+        plugin: PluginType.IMAGES,
+        tag: 'tagC',
+        sample: 111,
+      },
+    };
+    let timeSeriesData: TimeSeriesData = {
```

Comment:
Could you use buildTimeSeriesData() from webapp/metrcis/testing.ts?  It might also be worth adding an `overrides` argument to that function so you can also use it in other places in these tests.

Reply:
I think you should just reuse buildTimeSeriesData() rather than introducing new createCustomizedTimeSeriesData(). We do this commonly elsewhere in the codebase. You can make override an optional argument and none of the existing usages should notice the change.